tools/cldr-apps/src/main/java/org/unicode/cldr/web/ReportsDB.java

```diff
@@ -0,0 +1,83 @@
+package org.unicode.cldr.web;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.logging.Logger;
+
+import org.unicode.cldr.util.CLDRLocale;
+import org.unicode.cldr.util.VoterReportStatus;
+
+public class ReportsDB implements VoterReportStatus<Integer>, ReportStatusUpdater<Integer> {
+    public static ReportsDB getInstance() {
+        return ReportsDBHelper.INSTANCE;
+    }
+    static final Logger logger = SurveyLog.forClass(ReportsDB.class);
+    final String table = DBUtils.Table.VOTE_REPORTS
+        .forVersion(SurveyMain.getNewVersion(), false).toString();
+
+    final static class ReportsDBHelper {
+        static ReportsDB INSTANCE = new ReportsDB();
+    }
+
+    private void setupDB() {
+        logger.info("Setting up DB");
+        try {
+            try (Connection conn = DBUtils.getInstance().getDBConnection();) {
```

Comment:
just wondering, how to choose between getDBConnection [setAutoCommit(false)] and getAConnection?

Reply:
It's a different flow of the function.  Auto commit means you don't call commit() manually, non auto commit means you do.  For this function where I'm expecting to perform updates and then call one commit manually, I found this to work better.